checkov/terraform/graph_builder/local_graph.py

```diff
@@ -291,11 +291,13 @@ def _create_edge(self, origin_vertex_index: int, dest_vertex_index: int, label:
             return False
         edge = Edge(origin_vertex_index, dest_vertex_index, label)
         if cross_variable_edges:
-            if edge in self.edges or self.vertices[edge.dest].block_type != BlockType.RESOURCE or \
-                    self.vertices[edge.origin].block_type != BlockType.RESOURCE:
+            if self.vertices[dest_vertex_index].block_type != BlockType.RESOURCE or \
+                    self.vertices[origin_vertex_index].block_type != BlockType.RESOURCE:
+                return False
+            if edge in self.out_edges[origin_vertex_index]:
                 return False
             edge.label = CROSS_VARIABLE_EDGE_PREFIX + edge.label
-            if edge in self.edges:
+            if edge in self.out_edges[origin_vertex_index]:
```

Comment:
isn't this redundant, because you check it on line 297 already?

Reply:
no, because it's with a different label (with the cross variable)

Reply:
ah got it, yeah we compare the origin ID + dest ID + label for equality

Reply:
It is a bit misleading though, maybe create s a new variable and check it?
For example `edge_with_cross_variable_label_prefix`